HandEyeCalibration/UtilizeEyeInHandCalibration/UtilizeEyeInHandCalibration.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+Utilize the result of eye-in-hand calibration to transform (picking) point
+coordinates from the camera frame to the robot base frame.
+*/
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+#include <opencv2/core/core.hpp>
+#include <cmath>
+#include <iostream>
+#include <filesystem>
+
+Eigen::MatrixXd cv2eigen(const cv::Mat &cv_mat)
+{
+    if(cv_mat.dims > 2)
+    {
+        throw "Invalid matrix dimensions. Expected 2D.";
+    }
+
+    Eigen::MatrixXd eigen_mat(cv_mat.rows, cv_mat.cols);
+
+    for(int i = 0; i < cv_mat.rows; i++)
+    {
+        for(int j = 0; j < cv_mat.cols; j++)
+        {
+            eigen_mat(i, j) = cv_mat.at<double>(i, j);
+        }
+    }
+
+    return eigen_mat;
+}
+
+
+cv::Mat read_transform(const std::string file_name)
+{
+    auto file_storage = cv::FileStorage::FileStorage();
+
+    if(!file_storage.open(file_name, cv::FileStorage::Mode::READ))
+    {
+        std::cout << "Could not open file" << std::endl;
+    }
+    auto pose_state_node = file_storage["PoseState"];
+    if(pose_state_node.empty())
+    {
+        throw "PoseState not found in file";
+    }
+    auto pose_state = pose_state_node.mat();
+    file_storage.release();
+
+	return pose_state;
+}
+
+int main()
+{
+    try
+    {
+        std::cout.precision(2);
```

Comment:
What does this do?

Reply:
it sets the float output to a maximum of precision of 2, same as what np.set_printoptions(precision=2) does in Python. Nevertheless, I can remove this